ush/python/pygfs/task/nexus_emission.py

```diff
@@ -0,0 +1,458 @@
+
+            dsets = []
+            for index in indexes:
+                # list files for log
+                logger.info(f" - {files[index]}, {index}")
+
+                # now concatenate the files per day
+                if os.path.exists(files[index]) is False:
+                    break
+                ds = xr.open_dataset(files[index], decode_cf=False)
+
+                # update time coordinate
+                ds = ds.assign_coords(time=('time', [index]))
+
+                # set time units to reference start-date
+                ds.time.attrs['units'] = self.start_date.strftime('hours since %Y-%m-%d %H:00:00')
+
+                # append
+                dsets.append(ds)
+
+            # concatenate all the files for this day
+            if dsets is None:
+                break
+            else:
+                ds = xr.concat(dsets, dim='time')
+
+            encoding = {var: {"zlib": True, "complevel": 2} for var in ds.data_vars}
+            outname = f"{self.task_config.NEXUS_DIAG_PREFIX}.{day_str}.nc"
+            ds.to_netcdf(outname, format="NETCDF4", encoding=encoding)
+            logger.info(f"Wrote daily output: {outname}")
+
+        logger.info("NEXUS emission processing execute phase complete")
+
+    @logit(logger)
+    def finalize(self) -> None:
+        """Perform closing actions of the task.
+        Copy processed files from the DATA directory to COMOUT_CHEM_INPUT.
+
+        Returns
+        -------
+        None
+
+        Notes
+        -----
+        Only copies processed NEXUS files to the output directory.
+        Uses FileHandler for reliable file operations with logging
+        """
+        logger.info("Finalizing NEXUS emissions processing")
+
+        FileHandler(self.task_config.nexus_emission.data_out).sync()
+
+        logger.info("Chemical emissions finalization complete")
+
+
+def _write_txt_file(content: str, file_path: Union[str, os.PathLike]) -> None:
+    """Write content to a text file.
+
+    Parameters
+    ----------
+    content : str
+        Content to write to the file.
+    file_path : Union[str, os.PathLike]
+        Path where the file will be created.
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    If the directory does not exist, it will be created.
+    """
+    os.makedirs(os.path.dirname(file_path), exist_ok=True)
+    with open(file_path, 'w') as f:
+        f.write(content)
+
+
+def _get_day_indices(datetimes):
+    """
+    Group indices of datetimes by day, including midnight in both days.
+
+    Parameters
+    ----------
+    datetimes : list of datetime.datetime
+        List of datetime objects.
+
+    Returns
+    -------
+    dict
+        Dictionary mapping datetime.datetime (at midnight) to list of indices.
+        Each day includes all hours from 00:00 of that day through 00:00 of the next day,
+        and the midnight index is included in both days.
+    """
+    from collections import defaultdict
+    from datetime import timedelta
```

Comment:
Please move these to the top of the file and/or consolidate them with what is already at the top.